addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/__tests__/environment-sc-cidr-service.test.js

```diff
@@ -283,6 +331,264 @@ describe('EnvironmentScCidrService', () => {
+      // BUILD
+      const requestContext = {};
+      const params = {
+        id: 'testId',
+        updateRequest: [
+          {
+            protocol: 'tcp',
+            fromPort: 123,
+            toPort: 123,
+            cidrBlocks: ['1.1.1.1/32', '4.4.4.4/32'],
+          },
+        ],
+      };
+      environmentScService.mustFind = jest.fn(() => ({
+        id: 'testId',
+        createdBy: 'someUser',
+      }));
+      const currentIngressRules = [
+        {
+          protocol: 'tcp',
+          fromPort: 123,
+          toPort: 123,
+          cidrBlocks: [],
+        },
+      ];
+      const authorizeParams = {
+        GroupId: 'samplesecurityGroupId',
+        IpPermissions: [
+          {
+            FromPort: 123,
+            IpProtocol: 'tcp',
+            IpRanges: [{ CidrIp: '1.1.1.1/32' }, { CidrIp: '4.4.4.4/32' }],
+            ToPort: 123,
+          },
+        ],
+      };
+      const securityGroupId = 'samplesecurityGroupId';
+
+      environmentScService.getSecurityGroupDetails = jest.fn(() => ({
+        currentIngressRules,
+        securityGroupId,
+      }));
+      service.revokeSecurityGroupIngress = jest.fn();
+      service.authorizeSecurityGroupIngress = jest.fn();
+      service.getEc2Client = jest.fn(() => {
+        return {};
+      });
+
+      // OPERATE
+      await service.update(requestContext, params);
+
+      // CHECK
+      expect(service.authorizeSecurityGroupIngress).toHaveBeenCalledTimes(1);
+      expect(service.authorizeSecurityGroupIngress).toHaveBeenCalledWith({}, authorizeParams);
+      expect(service.revokeSecurityGroupIngress).not.toHaveBeenCalled();
+    });
+
+    it('should use the correct CIDR blocks to make revoke and authorize calls with overlapping CIDR block requests', async () => {
+      // BUILD
+      const requestContext = {};
+      const params = {
+        id: 'testId',
+        updateRequest: [
+          {
+            protocol: 'tcp',
+            fromPort: 123,
+            toPort: 123,
+            cidrBlocks: ['1.1.1.1/32', '2.2.2.2/32', '3.3.3.3/32'],
+          },
+        ],
+      };
+      environmentScService.mustFind = jest.fn(() => ({
+        id: 'testId',
+        createdBy: 'someUser',
+      }));
+      const currentIngressRules = [
+        {
+          protocol: 'tcp',
+          fromPort: 123,
+          toPort: 123,
+          cidrBlocks: ['1.1.1.1/32', '4.4.4.4/32'],
+        },
+      ];
+      const revokeParams = {
+        GroupId: 'samplesecurityGroupId',
+        IpPermissions: [{ FromPort: 123, IpProtocol: 'tcp', IpRanges: [{ CidrIp: '4.4.4.4/32' }], ToPort: 123 }],
+      };
+      const authorizeParams = {
+        GroupId: 'samplesecurityGroupId',
+        IpPermissions: [
+          {
+            FromPort: 123,
+            IpProtocol: 'tcp',
+            IpRanges: [{ CidrIp: '2.2.2.2/32' }, { CidrIp: '3.3.3.3/32' }],
```

Comment:
1.1.1.1/32 doesn't need to be included since it is already there?

Reply:
Correct.